### PR TITLE
replace validate_* calls with datatypes & minor fixes

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

**Comment by bastelfreak** [comment]
@jbondpdx updated it.

**Comment by jbondpdx** [comment]
Awesome, thank you. I get your logic changing it, but consistency.

File: manifests/init.pp [hunk 1]
```diff
@@ -46,22 +46,16 @@
...
-  $package               = undef,
-  $package_options       = undef,
-  $java_alternative      = undef,
-  $java_alternative_path = undef,
-  $java_home             = undef
+  String $distribution                                              = 'jdk',
+  Pattern[/present|installed|latest|^[.+_0-9a-zA-Z:~-]+$/] $version = 'present',
+  $package                                                          = undef,
+  Optional[Array] $package_options                                  = undef,
+  $java_alternative                                                 = undef,
```
Comment (binford2k) [inline manifests/init.pp]: Shouldn't these be `Optional[String]`?

Comment (bastelfreak) [inline manifests/init.pp]: The goal was to replace all the validate_* calls with datatypes, not to add datatypes to all parameters. I normally mention this, but somehow forgot it here. I added datatypes to all params in init.pp now.

**Pull request merged**